traffic_portal/app/src/common/api/TopologyService.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var TopologyService = function($http, ENV, locationUtils, messageModel) {
+
+	this.getTopologies = function(queryParams) {
+		return $http.get(ENV.api['root'] + 'topologies', { params: queryParams }).then(
```

Comment:
I guess this won't work w/ API 3.0 until this is also updated in TP. Updating TP wasn't included as part of the "add API 3.0 PR". @mitchell852 can we simply change `ENV.api['root']` to point to 3.0 in a different PR?

Reply:
You could do that, but you could also do the reverse of what we did for the config files preview pages: add a property to `ENV.api`. You could call it like `ENV.api['next']`.

I'm not particular which approach is chosen, but I thought I'd bring up another option.

Reply:
yeah, i got a PR coming to upgrade the api version but was trying to decide if it belongs in this PR....i guess so.

Reply:
Maybe it makes sense to change TP to use 3.0 everywhere in a separate PR, then this would be rebased onto that and can be updated to use the 3.0 variable from the separate PR?

Reply:
probably a good idea